utils/fhirpath/fhir_path_processor.bal

```diff
@@ -103,6 +103,82 @@ public isolated function evaluateFhirPath(map<json> fhirResource, string fhirPat
 
 }
 
+# Select the resource elements from the given FHIR resource.
+#
+# + fhirPaths - FhirPath expressions for the required elements
+# + fhirResource - Requested fhir resource
+# + return - Returns FHIR resource with the selected elements
+public isolated function selectResourceElements(string[] fhirPaths, map<json> fhirResource) returns map<anydata>|error {
+    map<json> tempFhirResource = fhirResource.clone();
+    string resourceType = check fhirResource["resourceType"].cloneWithType(string);
+
+    map<anydata> elementMap = {
+        "resourceType": {},
+        "meta": {}
+    };
+
+    foreach string fhirPath in fhirPaths {
+        string[] fhirPathTokens = regex:split(fhirPath, "\\.");
+        string fhirPathResourceType = fhirPathTokens[0].trim();
+        if resourceType == fhirPathResourceType {
+            map<anydata> parentElement = {};
+            parentElement = elementMap;
+            foreach string fhirPathToken in fhirPathTokens {
+                map<anydata> childElement = {};
+                string childElementName = fhirPathToken.trim();
+
+                if !parentElement.hasKey(childElementName) {
+                    parentElement[childElementName] = {};
+                }
+                childElement = <map<anydata>>parentElement[childElementName];
+                parentElement = childElement;
+            }
+        }
+    }
+
+    foreach [string, json] entry in tempFhirResource.entries() {
```

Comment:
`foreach [string, json] [x, y] in tempFhirResource.entries() {`
You can use the above pattern with nice variable names.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/35eb466e4284e621e7cb661d9d595e4d2eb8eabb